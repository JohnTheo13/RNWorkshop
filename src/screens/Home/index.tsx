import React from 'react';
import { Text, ImageBackground, StyleSheet, Button, View } from 'react-native';
import { NavigationStackProp } from 'react-navigation-stack';
import { white } from '../../config/theme';

const style = StyleSheet.create({
  header: {
    fontSize: 40,
    textAlign: 'center',
    color: white,
    marginTop: 20,
    flex: 1
  },
  image: {
    width: '100%',
    height: '100%'
  },
  button: {
    bottom: 20
  }
});

export default ({
  navigation: {
    state: { routeName },
    navigate
  }
}: {
  navigation: NavigationStackProp;
}): React.FunctionComponentElement<{}> => (
  <ImageBackground
    source={{ uri: 'https://facebook.github.io/react/logo-og.png' }}
    style={style.image}
  >
    <Text style={style.header}>
      Welcome to the RN workshop {routeName} screen
    </Text>
    <View style={style.button}>
      <Button onPress={(): boolean => navigate('Details')} title="To Details" />
    </View>
  </ImageBackground>
);
