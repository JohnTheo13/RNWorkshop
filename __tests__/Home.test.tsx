import React from 'react';
import { Button, Text } from 'react-native';
import renderer from 'react-test-renderer';
import Home from '../solution/screens/Home';
// Note: test renderer must be required after react-native.

it('renders correctly', () => {
  const navigation = {
    state: {
      routeName: 'Home'
    },
    navigate: jest.fn()
  };

  const component = renderer.create(<Home navigation={navigation} />);
  // console.log(component.root.findByType(Text).props.children[0]);

  expect(component.root.findAllByType(Text)[0].props.children.join('')).toBe(
    `Welcome to the RN workshop ${navigation.state.routeName} screen`
  );

  renderer.act(() => {
    component.root.findByType(Button).props.onPress();
  });

  expect(navigation.navigate).toHaveBeenCalledTimes(1);

  expect(component.toJSON()).toMatchSnapshot();
});
