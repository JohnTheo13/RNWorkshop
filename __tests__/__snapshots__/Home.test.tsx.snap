// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://facebook.github.io/react/logo-og.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 40,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Welcome to the RN workshop 
      Home
       screen
    </Text>
    <View
      style={
        Object {
          "bottom": 20,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            To Details
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
